akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -38,7 +38,15 @@ private class BackoffOnRestartSupervisor(
         // to our own Restarts, which involves stopping the child.
         case Restart ⇒
           val childRef = sender()
-          become(waitChildTerminatedBeforeBackoff(childRef) orElse handleBackoff)
+          val nextRestartCount = restartCount + 1
+          if (strategy.maxNrOfRetries >= 0 && nextRestartCount > strategy.maxNrOfRetries) {
```

Comment:
Sorry took me forever to get back here -- I assume you're referring to the "withinTimeRange" parameter to the OneForOneStrategy? I can definitely add that into this PR if desired.